code/graphics/software/font.cpp

```diff
@@ -590,12 +590,15 @@ namespace font
 				str[len] = ellipsis_char;
 			}
 
+			// measure with ellipsis before shrinking further
```

Comment:
Add a note to this comment, something like `(because the characters could be narrower than the characters they replaced)` to explain why